bindings/python/iota_sdk/types/common.py

```diff
@@ -12,21 +12,28 @@
 
 
 def json(cls):
-    """Decorator to add custom to_dict and to_json methods to a dataclass."""
-    # Apply the dataclass_json decorator first to get the default behavior
+    """Decorator to add to_dict and to_json methods to a dataclass."""
+
+    # Store override method
+    override_to_dict = getattr(cls, "to_dict", None)
+
+    # Apply the dataclass_json decorator to get the default behavior
     cls = dataclass_json(
         letter_case=LetterCase.CAMEL,
         undefined=Undefined.RAISE)(cls)
 
-    # Store original methods
-    original_to_dict = cls.to_dict
+    # Re-apply the original fn if it exists
+    if override_to_dict is not None:
+        setattr(cls, "_to_dict", override_to_dict)
+    else:
+        setattr(cls, "_to_dict", cls.to_dict)
```

Comment:
Don't really get what's the point of this, if `to_dict` exists, you set it for `_to_dict`, just to set it as `to_dict` again inside of `custom_to_dict()`?

Reply:
yea i think we can remove line28-29, and remove _ from _to_dict() and were golden

Reply:
The reason is because there are two different definitions:
1. the fn that converts to a dict
2. the wrapper fn that calls the above and removes `None` values

Because of that, there must be two different fns in the class.

Reply:
This here seems to just work fine
```Python
def json(cls):
    """Decorator to add to_dict and to_json methods to a dataclass."""

    # Get potential override method
    to_dict = getattr(cls, "to_dict", None)

    # Apply the dataclass_json decorator to get the default behavior
    cls = dataclass_json(
        letter_case=LetterCase.CAMEL,
        undefined=Undefined.RAISE)(cls)

    # If no custom one is defined, set the default from dataclass_json
    if to_dict is None:
        to_dict = cls.to_dict

    # Override to_dict to remove None values
    def custom_to_dict(self, *args, **kwargs):
        # pylint: disable=protected-access
        original_dict = to_dict(self, *args, **kwargs)

        result = {k: v for k, v in original_dict.items() if v is not None}
        return result

    def custom_to_json(self, *args, **kwargs):
        # Use the custom to_dict method for serialization
        return DataClassJsonMixin.to_json(self, *args, **kwargs)

    setattr(cls, "to_dict", custom_to_dict)
    setattr(cls, "to_json", custom_to_json)

    return cls
```

Reply:
fair enough. python is weird